### ISSUE TITLE
avoid keyboard coming up when closing coordinates input

### ISSUE THREAD

**Issue by Bananeweizen** [cgeo/cgeo#5048]
from facebook: when you close the coordinates input, the bearing field gets the focus and leads to the keyboard popping up

**Comment by Bananeweizen** [comment]
I have no idea how to fix this. I tried giving the initial focus to the buttons instead of the bearing text, I added a non visible layout above the edit text, I disabled soft input on the edittext and several other things you find on stackoverflow. None of that actually avoided the keyboard coming up, when coming back to this activity from the coords dialog.

Anyone with other ideas? Something to be changed in the layout of the activity itself to no longer focus the bearing?

**Comment by Lineflyer** [comment]
Isn't there any default for each activity also given in `androidmanifest.xml`? I have seen alike while doing changes there (but have no idea what they mean).

**Comment by Bananeweizen** [comment]
Yes, you can have _another_ control take the focus, like the topmost button instead of the text input. But even then the keyboard comes up, and it is even less useful there. :)

**Comment by chris2016** [comment]
I'm not yet through all the code but it is not related to the onFocus of the bearing edittext. You can change it and then it will never popup the keyboard onFocus. It is only coming after the dismiss dialog function. 

@Override
    public void updateCoordinates(final Geopoint gp) {
        buttonLat.setText(gp.format(GeopointFormatter.Format.LAT_DECMINUTE));
        buttonLon.setText(gp.format(GeopointFormatter.Format.LON_DECMINUTE));


final Handler closeKeyboardHandler = new Handler();
        closeKeyboardHandler.postDelayed(new Runnable() {

            @Override
            public void run() {
                
                if (bearing != null) {
                    final InputMethodManager imm = (InputMethodManager) getSystemService(
                                    Context.INPUT_METHOD_SERVICE);
                    imm.hideSoftInputFromWindow(
                            bearing.getWindowToken(), 0);
                }
            }
        }, 100);

...


This will close the keyboard but not preventing the keyboard from showing up. I will have a further look into it,

**Comment by rsudev** [comment]
@chris2016  Thanks for chiming in. I'll assign it 'virtually' to you (setting myself as stand-in)

**Comment by chris2016** [comment]
I found a solution. 

1. In the onCreate method of EditWaypointActivity call   getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_HIDDEN);

This prevents the Keyboard showing up if the focus on the Popup was changed. Unfortunately if you dont change the focus e.g. select lat or lon box than the keyboard still appears(Just open the Dialog and close it without changing the focus)

2. In CordiantesInputDialog set the focus to a visible edittext when the dialog opens:

e.g.
eLat.requestFocus();
                if(eLat.requestFocus()) {
                    getActivity().getWindow().setSoftInputMode(WindowManager.LayoutParams.SOFT_INPUT_STATE_ALWAYS_VISIBLE);
                }

After that the Edittext disappears when closing the dialog. Setting the focus seems to be a workaround but I could'nt find another way. Maybe someone has an Idea why it is depending on the focus change.

BR
Chris
